blockchain/chain/src/store/chain_store.rs

```diff
@@ -964,18 +951,33 @@ mod tests {
             .miner_address(Address::new_id(0))
             .build()
             .unwrap();
+        let genesis_ts = Tipset::new(vec![gen_block.clone()]).unwrap();
+        let cs = ChainStore::new(db, chain_config, genesis_ts.clone());
 
         assert_eq!(cs.genesis().unwrap(), None);
         cs.set_genesis(&gen_block).unwrap();
+        cs.set_heaviest_tipset(Arc::new(genesis_ts)).unwrap();
         assert_eq!(cs.genesis().unwrap(), Some(gen_block));
     }
 
     #[test]
     fn block_validation_cache_basic() {
         let db = forest_db::MemoryDB::default();
         let chain_config = Arc::new(ChainConfig::default());
+        let gen_block = BlockHeader::builder()
+            .epoch(1)
+            .weight(2_u32.into())
+            .messages(Cid::new_v1(DAG_CBOR, Identity.digest(&[])))
+            .message_receipts(Cid::new_v1(DAG_CBOR, Identity.digest(&[])))
+            .state_root(Cid::new_v1(DAG_CBOR, Identity.digest(&[])))
+            .miner_address(Address::new_id(0))
+            .build()
+            .unwrap();
+        let genesis_ts = Tipset::new(vec![gen_block.clone()]).unwrap();
 
-        let cs = ChainStore::new(db, chain_config);
+        let cs = ChainStore::new(db, chain_config, genesis_ts.clone());
+        cs.set_genesis(&gen_block).unwrap();
+        cs.set_heaviest_tipset(Arc::new(genesis_ts)).unwrap();
```

Comment:
You're adding 12 lines here. Could you explain them to me?

Reply:
Oh, I think I see what you're trying to do. You can simplify this a lot. Either use a simpler genesis block or use `read_genesis_header` with the calibnet genesis data.

Reply:
Simplified.